src/lsp/cobol_ptree/branching_statements.ml

```diff
@@ -21,52 +21,38 @@ open Simple_statements
 
 (* --- *)
 
-(* GOTO <target> *)
+(* GOTO *)
 type goto_stmt =
-  {
-    goto_target: procedure_name with_loc;
-  }
-[@@deriving ord]
-
-let pp_goto_stmt ppf { goto_target } =
-  Fmt.pf ppf "@[GO TO @[%a@]" pp_procedure_name' goto_target
-
-(* GOTO ENTRY *)
-type goto_entry =
-  {
-    goto_entry_target: alphanum_string
-  }
-[@@deriving ord]
-
-let pp_goto_entry ppf { goto_entry_target } =
-  Fmt.pf ppf "@[GO TO @[%a@]" pp_alphanum_string goto_entry_target
-
-(* GOTO DEPENDING *)
-type goto_depending_stmt =
-  {
-    goto_depending_targets: procedure_name with_loc nel;
-    goto_depending_on: ident;
-  }
+  | GoToSimple of
+    {
+      goto_targets: procedure_name with_loc nel;
+      goto_depending_on: ident option;
+    }
+  | GoToEntry of
+    {
+      goto_entry_targets: alphanum_string with_loc nel;
```

Comment:
That should not be `alphanum_string`; please use `alphanum` instead.

Reply:
The reason I used `alphanum_string` is because no comparison function for `alphanum` is created. I'm not sure how I should deal with that

Reply:
Yes you can do it like this.